### PR TITLE
Fix authentication and new keyword errors

### LINKED ISSUE #100: Food supply Quorum tutorial error "authentication needed: password or unlock"
@ThomasRalee while working updating the docs to reflect the protected Quorum APIs, I tested the [Quorum food supply tutorial](https://docs.dev.chainstack.com/tutorials/food-supply-temperature-control-on-quorum) and it's not deploying anymore.

Note that I specify the `.env` in the following format (just in case, I don't think this is the issue):

```
RPC1='https://user-name:pass-word-pass-word-pass-word@nd-123-456-789.int.chainstack.com'
PK1='TRANSACTION_MANAGER_PUBLIC_KEY'
RPC2='https://user-name:-pass-word-pass-word-pass-word@nd-123-456-789.int.chainstack.com'
PK2='TRANSACTION_MANAGER_PUBLIC_KEY'
RPC3='https://user-name:pass-word-pass-word-pass-word@nd-123-456-789.int.chainstack.com'
PK3='TRANSACTION_MANAGER_PUBLIC_KEY'
```

Whenever I run `node public.js` it's failing with:

```
(node:8646) UnhandledPromiseRejectionWarning: Error: Returned error: authentication needed: password or unlock
    at Object.ErrorResponse (/root/fuck/node_modules/web3-core-helpers/src/errors.js:29:16)
    at /root/fuck/node_modules/web3-core-requestmanager/src/index.js:140:36
    at XMLHttpRequest.request.onreadystatechange (/root/fuck/node_modules/web3-providers-http/src/index.js:96:13)
    at XMLHttpRequestEventTarget.dispatchEvent (/root/fuck/node_modules/xhr2-cookies/dist/xml-http-request-event-target.js:34:22)
    at XMLHttpRequest._setReadyState (/root/fuck/node_modules/xhr2-cookies/dist/xml-http-request.js:208:14)
    at XMLHttpRequest._onHttpResponseEnd (/root/fuck/node_modules/xhr2-cookies/dist/xml-http-request.js:318:14)
    at IncomingMessage.<anonymous> (/root/fuck/node_modules/xhr2-cookies/dist/xml-http-request.js:289:61)
    at IncomingMessage.emit (events.js:214:15)
    at endReadableNT (_stream_readable.js:1178:12)
    at processTicksAndRejections (internal/process/task_queues.js:77:11)
(node:8646) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:8646) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
Error: Returned error: authentication needed: password or unlock
    at Object.ErrorResponse (/root/fuck/node_modules/web3-core-helpers/src/errors.js:29:16)
    at /root/fuck/node_modules/web3-core-requestmanager/src/index.js:140:36
    at XMLHttpRequest.request.onreadystatechange (/root/fuck/node_modules/web3-providers-http/src/index.js:96:13)
    at XMLHttpRequestEventTarget.dispatchEvent (/root/fuck/node_modules/xhr2-cookies/dist/xml-http-request-event-target.js:34:22)
    at XMLHttpRequest._setReadyState (/root/fuck/node_modules/xhr2-cookies/dist/xml-http-request.js:208:14)
    at XMLHttpRequest._onHttpResponseEnd (/root/fuck/node_modules/xhr2-cookies/dist/xml-http-request.js:318:14)
    at IncomingMessage.<anonymous> (/root/fuck/node_modules/xhr2-cookies/dist/xml-http-request.js:289:61)
    at IncomingMessage.emit (events.js:214:15)
    at endReadableNT (_stream_readable.js:1178:12)
    at processTicksAndRejections (internal/process/task_queues.js:77:11)
```

This probably has to do with [web3 version 1.2.1](https://www.npmjs.com/package/web3).

I'm assigning this issue to you for the time being. Please let me know how this can be fixed and I'll update the tutorial.

### PR DESCRIPTION
**Pull request by syahrul12345**
Fixes #100 

@akegaviar Please try the new script `public.js`. Let me know if it still doesnt work. For `private.js`, I've just added in unlocking the node account.

### CODE REVIEW

**Comment by easeev** [comment]
@syahrul12345,

1. Please create PRs with a clear title and description. What does the introduced code do?
2. Did you update https://github.com/chainstack/quorum-iot-tutorial?

File: docs/tutorials/food-supply-temperature-control-on-quorum.md [hunk 1]
```diff
@@ -212,15 +213,17 @@ async function deployContract(web3) {
...
+    from: address
+  }).then((receipt) => {
     return receipt.status;
   });
 }
 
 async function getTemperature(web3, contractAddress) {
   const myContract = await getContract(web3, contractAddress);
-
+  const address = await getAddress(web3);2
```
Comment (easeev) [inline docs/tutorials/food-supply-temperature-control-on-quorum.md]: Typo?

**Comment by syahrul12345** [comment]
`public.js` has been changed where all functions no longer call `web3.eth.methods.MyMethod()`. Instead:
1) I use web3 to encode the function call into bytecode using web3.eth.
2) Include the bytecode in the transaction payload
4) Sign the `txPayload`
3) Send it via `web3.eth.sendSignedTransaction(signedTxPayload)`

File: docs/tutorials/food-supply-temperature-control-on-quorum.md [hunk 2]
```diff
@@ -85,10 +85,13 @@ In your project folder, create an `.env` file:
 ``` js
 RPC1='RPC_ENDPOINT'
 PK1='CONSTELLATION_PUBLIC_KEY'
+PRIV1= 'YOUR_WALLET1_PRIVATE_KEY'
```
Comment (easeev) [inline docs/tutorials/food-supply-temperature-control-on-quorum.md]: Can we use a mnemonic phrase instead?

**Comment by easeev** [comment]
It looks like now the accounts that are used for public and private transactions are different (for public we use the one that we init via private key and for private the one on the node). Isn't it confusing? Since we're keeping hosted accounts on the nodes for now, why can't we keep using the hosted one?

File: docs/tutorials/food-supply-temperature-control-on-quorum.md (hunk 2 above)
Comment (syahrul12345) [reply docs/tutorials/food-supply-temperature-control-on-quorum.md]: Nope, signing via web3 requires privatekey only. If we want to use mnemonic, we'll need to use a hd-wallet like truffle's hd-wallet-provider. Still i will do research on this.

**Comment by syahrul12345** [comment]
@ake @easeev I tried this on prod, it works fine. I'm using node hosted accounts for both public and private. 

1) To prevent getting authentication errors, function calls now implement `await web3.eth.personal.unlockAccount(address,'',1000)`. This is for both private and public

2) For private transctions,`.call()` now throws an `Return values invalid` instead of `null` when we are calling to a private contract that we are not party to with Web3 1.2. I've added in a `catch` hook in the `getTemperature()` function

```
async function getTemperature(web3, contractAddress) {
  const myContract = await getContract(web3, contractAddress);
  await web3.eth.personal.unlockAccount(address,'',1000)
  return myContract.methods.get().call()
  .then(result => result)
  .catch(error => null)
}
```

**Comment by easeev** [comment]
@syahrul12345 please sync with https://github.com/chainstack/quorum-iot-tutorial/pull/1
@akegaviar please verify that it works and the initial issue is solved

**Comment by akegaviar** [comment]
I ran the instructions in the updated doc.

Bot `public.js` and `private.js` deployments go through.

`public.js` throws no errors.

running the `public.js` one however prints the following error:

```
(node:8701) UnhandledPromiseRejectionWarning: ReferenceError: address is not defined
    at getTemperature (/root/food/private.js:102:41)
    at processTicksAndRejections (internal/process/task_queues.js:85:5)
    at async main (/root/food/private.js:41:16)
(node:8701) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:8701) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
privateJSerror

**Comment by syahrul12345** [comment]
@akegaviar my apologies, i've added in the address function call. Should work now!

**Comment by akegaviar** [comment]
Thanks @syahrul12345, it works now:

```
Contract address after deployment: 0x5F67055f511741c7aEF7Ffc4B706479586EbFcA2
[Node3] temp retrieved after updating contract from external nodes: null
[Node2] temp retrieved after updating contract from internal nodes: 12
[Node3] temp retrieved from external nodes after update null
```

**Pull request merged**